recipe/meta.yaml

```diff
@@ -12,23 +11,38 @@ source:
 
 build:
   number: 0
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<37] 
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --no-cache --ignore-installed -vvv
   entry_points:
     - install_cmdstan = cmdstanpy.install_cmdstan:main
     - install_cxx_toolchain = cmdstanpy.install_cxx_toolchain:main
 
 requirements:
   host:
     - pip
-    - python >=3.7
+    - python
+    - setuptools
+    - wheel
   run:
-    - python >=3.7
+    - python
     - pandas
     - numpy >=1.21
     - cmdstan
     - tqdm
+  run_constrained:  # [osx]
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.14") }}  # [osx and x86_64]
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("11.1") }}  # [osx and arm64]
 
+test:
+  imports:
+    - cmdstanpy
+    - cmdstanpy.stanfit
+    - cmdstanpy.utils
+  requires:
+    - pip
+  commands:
+    - export CONDA_BUILD_SYSROOT=/opt/MacOSX10.14.sdk  # [osx and x86_64]
+    - export CONDA_BUILD_SYSROOT=/opt/MacOSX11.1.sdk  # [osx and arm64]
```

Comment:
Test commands won't work if the file run_test.py exists (the linter says it too). Your approach of adding env variables inside run_test.py was correct

Reply:
As we discussed in slack, I think setting env variables in the run_test.py script is not correct since the users of osx-64 would need to do the same in their environment before they can use cmdstanpy/cmdstan, which is not great. Hence, perhaps fixing/enhancing the activation script of cmdstan is the way forward.